src/interned.rs

```diff
@@ -391,7 +392,9 @@ where
 
         // The slot is valid in this revision but we have to sync the value's revision.
         let current_revision = zalsa.current_revision();
-        value.last_interned_at.store(current_revision);
+        if value.last_interned_at.load() < current_revision {
+            value.last_interned_at.store(current_revision);
+        }
 
```

Comment:
You could use `fetch_max` here to store the maximum between the current revision and the `last_interned_at` to avoid two separate atomic operations.

Reply:
This is `AtomicRevision`, not `AtomicUsize`. I will need to define `fetch_max()`, and it's not worth it. It's not like a race condition is problematic here.

Reply:
`AtomicRevision` is just a small wrapper around `AtomicUsize`. You can see `OptionalAtomicRevision` how we exposed other atomic methods.

This isn't just about races, it's also about avoiding unnecessary atomic operations

Reply:
`fetch_max()` won't be any faster; it needs to be an atomic RMW. Even on x86, [it compiles to a `cmpxchg` loop](https://godbolt.org/z/8c31as8oG), compared to load+store that compiles to normal instructions.

Reply:
Although we can be faster via branchless; I will change to that.